XenAdmin/Controls/DataGridViewEx/CollapsingPoolHostDataGridViewRow.cs

```diff
@@ -125,11 +126,18 @@ public bool HasPool
         public void SetCollapseIcon()
         {
             _expansionCell.Value = Resources.tree_minus;
+            _isACollapsedRow = false;
         }
 
         public void SetExpandIcon()
         {
             _expansionCell.Value = Resources.tree_plus;
+            _isACollapsedRow = true;
+        }
+
+        public bool IsACollapsedRow

```

Comment:
Since there is no logic in the property body, you can make the code more compact by defining it as:
```
 public bool IsACollapsedRow { get; protected set; }
```
and then you don't need the extra field.